### ISSUE TITLE
Enabling doesn't work after restart

### ISSUE THREAD

**Issue by rapic-dev** [MrMEEE/ironhide#68]
Hey guys! I'm having some problems with enabling the card after restart. First I've done a manual configuration with ironhide-configuration and restarted my laptop. After that I've edited the ironhide-disablecard and ironhide-enablecard from the examples for Dell Inspiron N5110. Then checked it with optirun glxgears and it worked fine. Now, after a restart with those settings I'm getting this when running glxgears:

$ optirun glxgears
- Starting Ironhide X server ironhide
  _PS0 Enabling nVidia card succeeded.
  FATAL: Error inserting nvidia_current (/lib/modules/2.6.38-11-generic/updates/dkms/nvidia-current.ko): No such device
  ... \* The nvidia driver was not loaded, try: sudo modprobe nvidia-current
- The Ironhide Xserver failed to start. Please check /var/log/Xorg.8.log[fail] 
  ironhide could not be started - optirun cannot be executed.

Also I tried to change "Always enable the nVidia card before disabling" to NO but nothing changed.

Any suggestions?

Cheers!

**Comment by MrMEEE** [comment]
please try to:

comment out : $ENABLECARD and $DISABLECARD at line 16 and 18 in /etc/pm/power.d/ironhide-disablecard-on-powerup

and reboot.. 

If that works, we can go from there..

**Comment by rapic-dev** [comment]
I'm getting the same error again, but now I can hear the fan spinning. Before commenting out those 2 lines the fan was dead silent. Looks like the card is now enabled after powerup but the error is still present.

**Comment by MrMEEE** [comment]
ok.. try:

update-rc.d -f ironhide remove

and reboot

**Comment by rapic-dev** [comment]
Ok, there are some results. After reboot the first time I run optirun glxgears it works without a problem, but on the second run I'm getting the exact same error from the beginning. Maybe there is a problem with the enable/disable scripts? As I said I'm using those from examples for Dell Inspiron N5110.

**Comment by rapic-dev** [comment]
Also I tried to run ./ironhide-disablecard and ./ironhide-enablecard on there own.

Here's what I get when enabling:
_PS0 Enabling nVidia card succeeded.
FATAL: Error inserting nvidia_current (/lib/modules/2.6.38-11-generic/updates/dkms/nvidia-current.ko): No such device

and for disabling:
ERROR: Module nvidia does not exist in /proc/modules
_DSM Disabling nVidia card succeeded.
_PS3 Disabling nVidia card succeeded.

When enabled the fan starts to work and when disabled it quiets down. So the hardware part of it works.

**Comment by MrMEEE** [comment]
Could you please try (if you haven't already), to run the ironhide-configuration..

First try using "Branko Vukelic"s configuration and then "splinter@splinter.pl"s configuration...

and see if there is a differens between these two configurations...

**Comment by rapic-dev** [comment]
Yes, looks like there are differences between those configurations, here are the results:

"splinter@splinter.pl"s config
enabling works, but on disable it says:
- Stopping Ironhide X server ironhide  
  ERROR: Module nvidia_current does not exist in /proc/modules
  Error: could not unload nvidia module, leaving card turned on

"Branko Vukelic"s config
enabling:
- Starting Ironhide X server ironhide  
  _PS0 Enabling nVidia Card failed (Error: AE_NOT_FOUND).

disabling:
- Stopping Ironhide X server ironhide  
  NVOP Disabling nVidia Card failed (Error: AE_NOT_FOUND).
  _PS3 Disabling nVidia Card failed (Error: AE_NOT_FOUND).

**Comment by rapic-dev** [comment]
After much testing I've found out that the "_SB.PCI0.PEG0.PEGP._OFF" and "_SB.PCI0.PEG0.PEGP._ON" acpi_call commands work best on my system. I think the main problem is when the card is disabled it won't enable again properly. The card itself is being enabled, I can hear the fan starts working and the power consumption raises, but it won't load the drivers. When enabling again I'm always getting the "FATAL: Error inserting nvidia_current (/lib/modules/2.6.38-11-generic/updates/dkms/nvidia-current.ko): No such device" error.

**Comment by iskeledz** [comment]
I am having the exact same issue on my Dell Inspiron N7110 (the 17" model) and I can confirm that the issue is hardware specific. I am very new to Linux, so could you please be a bit more specific on how to fix it (I didn't really understand your last post)? Does changing those commands completely remove the problem?  Also, the laptop runs glxgears in non-optirun mode after I type optirun glxgears. I have also tried it with another app (Neverball) - same problem. Thanks in advance.

**Comment by rapic-dev** [comment]
Sadly I still can't get ironhide to work properly on my system. The first acpi_call command from above just turns off the discrete graphics card so the battery lasts much longer. The other one turns on the graphics card but the drivers won't load or something like that. For some reason ironhide can't properly enable the card once it's disabled. Maybe Martin has some good tips in his pocket. :)

**Comment by iskeledz** [comment]
I would really like this to work since I am trying to switch from Windows to Linux. Ironhide is the only thing that worked so far (at least for one boot) - I couldn't configure Bumblebee properly in my previous Linux install. I also forgot to mention that I'm using Linux Mint 11.

**Comment by iskeledz** [comment]
I finally found a fix for my really dumb issue - economy mode was turned on and I was running on battery the whole time (duh). As for the fatal error, it seems to be gone after I changed my configuration and restarted a few times (you might even want to try my N7110 config). It's been running stable (no errors) for already 3 days. I'm using nvidia-current drivers from the xorg-edgers PPA. Hope it helps.
